### PR TITLE
update Travis CI configuration

### PR DESCRIPTION
**Pull request by xabbuh**
* use container based infrastructure
* add PHP 7 build job
* cache Composer cache directory on Travis
* report build status as fast as possible
* add build with lowest possible dependency versions
* run tests with different Symfony versions

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,37 @@
...
+      env: SYMFONY_VERSION="2.3.x"
+    - php: 5.6
+      env: SYMFONY_VERSION="2.6.x"
+    - php: 5.6
+      env: SYMFONY_VERSION="2.7.x"
+    - php: 5.6
+      env: SYMFONY_VERSION="2.8.x-dev"
+
+before_install:
+    - if [ "$SYMFONY_VERSION" != "" ]; then composer require --dev --no-update symfony/symfony:"$SYMFONY_VERSION"; fi
```
Comment (stof) [inline .travis.yml]: you should do that after updating composer itself (or not update composer)

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
     - composer self-update
-    - composer install --prefer-source
+
+install:
+    - if [ "$deps" = "low" ]; composer update --prefer-lowest; else composer install --prefer-source; fi
```
Comment (stof) [inline .travis.yml]: ``--prefer-source`` should be removed, as it kills all benefits of persisting the cache (downloaded archives go in the cache, but not source installs)

File: .travis.yml (hunk 1 above)
Comment (Seldaek) [inline .travis.yml]: Seems fine to me as it's done with a --no-update

File: .travis.yml (hunk 2 above)
Comment (Seldaek) [inline .travis.yml]: Missed that, I'll fix directly

Comment (xabbuh) [inline .travis.yml]: Oh indeed, thanks for catching this @stof

**Pull request merged**